util/Migrator/DbScripts_transition/2024-01-10_00_AllOrgsEnableCollectionEnhancements.sql

```diff
@@ -0,0 +1,44 @@
+-- This script will enable collection enhancements for all organizations.
+
+-- Step 1: Insert into a temporary table with an additional column for batch processing, update 50 k at a time
+SELECT [Id] AS [OrganizationId], CAST(ROW_NUMBER() OVER(ORDER BY [Id]) / 50000 AS INT) AS Batch
+INTO #TempOrg
+FROM [dbo].[Organization];
+-- WHERE [FlexibleCollections] = 0;
+
+-- Step 2: Get the maximum batch number
+DECLARE @MaxBatch INT = (SELECT MAX(Batch) FROM #TempOrg);
+DECLARE @CurrentBatch INT = 0;
+
+-- Step 3: Process each batch
+WHILE @CurrentBatch <= @MaxBatch
+BEGIN
+    -- Execute the stored procedure for each OrganizationId in the current batch
+    DECLARE @OrganizationId UNIQUEIDENTIFIER;
+
+    DECLARE OrgCursor CURSOR FOR
+    SELECT [OrganizationId]
+    FROM #TempOrg
+    WHERE [Batch] = @CurrentBatch;
+
+    OPEN OrgCursor;
+
+    FETCH NEXT FROM OrgCursor INTO @OrganizationId;
+
+    WHILE @@FETCH_STATUS = 0
+    BEGIN
+        -- Execute the stored procedure for the current OrganizationId
+        EXEC [dbo].[Organization_EnableCollectionEnhancements] @OrganizationId;
```

Comment:
As discussed, not sure if this will perform well enough to run for the whole db, but will defer to @rkac-bw on that one

Reply:
That is a valid concern but I thought it wouldn't be a problem since we're applying the changes per each `OrganizationId`

Reply:
I think this one will do well as it is batched and per org